meilisearch-http/src/routes/tasks.rs

```diff
@@ -536,65 +536,149 @@ fn filter_out_inaccessible_indexes_from_query<const ACTION: u8>(
     query
 }
 
-/// Deserialize a datetime optional string using rfc3339, assuming midnight and UTC+0 if not specified
-pub mod rfc3339_date_or_datetime {
-    #[allow(clippy::wildcard_imports)]
-    use super::*;
-    use serde::Deserializer;
-    use time::format_description::well_known::iso8601::{Config, EncodedConfig};
-    use time::format_description::well_known::{Iso8601, Rfc3339};
-    use time::{Date, PrimitiveDateTime, Time};
-    const SERDE_CONFIG: EncodedConfig = Config::DEFAULT.set_year_is_six_digits(true).encode();
-
-    /// Deserialize an [`Option<OffsetDateTime>`] from its ISO 8601 representation.
-    pub fn deserialize<'a, D: Deserializer<'a>>(
-        deserializer: D,
-    ) -> Result<Option<OffsetDateTime>, D::Error> {
-        deserializer.deserialize_option(Visitor)
+pub(crate) mod date_deserializer {
+    use time::{
+        format_description::well_known::Rfc3339, macros::format_description, Date, Duration,
+        OffsetDateTime, Time,
+    };
+
+    enum DeserializeDateOption {
+        Before,
+        After,
     }
-    struct Visitor;
 
-    #[derive(Debug)]
-    struct DeserializeError;
+    fn deserialize_date<E: serde::de::Error>(
+        value: &str,
+        option: DeserializeDateOption,
+    ) -> std::result::Result<OffsetDateTime, E> {
+        // We can't parse using time's rfc3339 format, since then we won't know what part of the
+        // datetime was not explicitly specified, and thus we won't be able to increment it to the
+        // next step.
+        if let Ok(datetime) = OffsetDateTime::parse(value, &Rfc3339) {
+            // fully specified up to the second
+            // we assume that the subseconds are 0 if not specified, and we don't increment to the next second
+            Ok(datetime)
+        } else if let Ok(datetime) = Date::parse(
+            value,
+            format_description!("[year repr:full base:calendar]-[month repr:numerical]-[day]"),
+        ) {
+            let datetime = datetime.with_time(Time::MIDNIGHT).assume_utc();
+            // add one day since the time was not specified
+            match option {
+                DeserializeDateOption::Before => Ok(datetime),
+                DeserializeDateOption::After => {
+                    let datetime = datetime
+                        .checked_add(Duration::days(1))
+                        .ok_or(serde::de::Error::custom("date overflow"))?;
+                    Ok(datetime)
+                }
+            }
+        } else {
+            Err(serde::de::Error::custom(
+                "could not parse a date with the RFC3339 or YYYY-MM-DD format",
```

Comment:
Did you notice [that YYYY-MM-DD is part of both](https://ijmacd.github.io/rfc3339-iso8601/)?

Reply:
The `time` crate only parses a fully specified `YYYY-MM-DDTHH:MM:SS(Z|[+-]HH:MM)` for rfc3339, which seems to be the closely interpretation of the specification ( https://tools.ietf.org/html/rfc3339#section-5.6 ) to me.

Reply:
But yes, I saw the Figma document and believe that the implementation of the deserialiser follows it. Regarding the error message, maybe we should say "a full date and time with the RFC3993 format or a date with YYYY-MM-DD format".

Reply:
No it is ok, I was just wondering if you saw the document or you were doing that by memory :)